File: llm_generated_code/Gemini/CPP/Level1/Q25.cpp
#include <iostream>
#include <string>

using namespace std;

bool isMirrorChar(char c) {
    return c == 'A' || c == 'H' || c == 'I' || c == 'M' || c == 'O' || c == 'T' || c == 'U' || c == 'V' || c == 'W' || c == 'X' || c == 'Y';
}

int main() {
    string word;
    cin >> word;

    int left = 0, right = word.length() - 1;
    bool isMirror = true;

    while (left < right) {
        if (!isMirrorChar(word[left]) || !isMirrorChar(word[right]) || word[left] != word[right]) {
            isMirror = false;
            break;
        }
        left++;
        right--;
    }

    cout << (isMirror ? "YES" : "NO");
    return 0;
}

File: llm_generated_code/Gemini/Python/Level1/Q25.py
def is_mirror_word(word):
  mirror_chars = {'A': 'A', 'H': 'H', 'I': 'I', 'M': 'M', 'O': 'O', 'T': 'T', 'U': 'U', 'V': 'V', 'W': 'W', 'X': 'X', 'Y': 'Y'}
  left, right = 0, len(word) - 1
  while left < right:
    if word[left] not in mirror_chars or word[right] not in mirror_chars or mirror_chars[word[left]] != word[right]:
      return "NO"
    left += 1
    right -= 1
  return "YES"

def main():
  word = input()
  print(is_mirror_word(word))

if __name__ == "__main__":
  main()
